Automattic/wp-calypso
# State: Add actions & data layer for saving Jetpack site type and vertical

Depends on D24416-code that introduces the new endpoints for saving site type and vertical.

#### Changes proposed in this Pull Request

* Introduce actions and data layer for saving Jetpack site type and topic.

#### Testing instructions

* Checkout this branch.
* Follow the first 3 steps of the testing instructions in D24416-code.
* Load `http://calypso.localhost:3000/jetpack/connect`.
* Pick a WP.com site and get its ID.
* Open your browser console.
* Type: 

```
dispatch( { type: 'JETPACK_CONNECT_SAVE_SITE_TYPE', siteId: 123456, siteType: 'business' } )
```

where `123456` is the ID of your site, and `business` is the site type.
* Type: 

```
dispatch( { type: 'JETPACK_CONNECT_SAVE_SITE_VERTICAL', siteId: 123456, siteVertical: 'test' } )
```

where `123456` is the ID of your site, and `business` is the site type.
* Open `/wp-admin/options.php` of your WP.com site.
* Verify you can find the `site_segment` option and it contains `business` as a value.
* Verify you can find the `site_vertical` option and it contains `test` as a value.
